src/app/common/search-input/search-input.component.ts

```diff
@@ -1,24 +1,69 @@
+import { animate, style, transition, trigger } from '@angular/animations';
 import { Component, EventEmitter, Input, Output } from '@angular/core';
 
+import { untilDestroyed, UntilDestroy } from '@ngneat/until-destroy';
+import { Subject } from 'rxjs';
+import { debounceTime } from 'rxjs/operators';
+
+@UntilDestroy()
 @Component({
 	selector: 'asy-search-input',
 	templateUrl: './search-input.component.html',
-	styleUrls: ['./search-input.component.scss']
+	styleUrls: ['./search-input.component.scss'],
+	animations: [
+		trigger('minCount', [
+			transition(':enter', [
+				style({ opacity: 0, height: 0 }),
+				animate('200ms ease', style({ opacity: 1, height: '*' }))
+			]),
+			transition(':leave', [
+				style({ opacity: 1, height: '*' }),
+				animate('200ms ease', style({ opacity: 0, height: 0 }))
+			])
+		])
+	]
 })
 export class SearchInputComponent {
 	@Input() placeholder = 'Search...';
 	@Output() readonly applySearch: EventEmitter<string> = new EventEmitter();
 	@Input() search = '';
+	@Input() width = '350px';
```

Comment:
@nrwinner should we be using null or `auto` as a default instead? Seems `350px` is very specific.